src/routes/api.rs

```diff
@@ -170,6 +170,15 @@ pub fn config(cfg: &mut web::ServiceConfig, rate_limit: &RateLimit) {
         web::resource("/admin/add")
           .wrap(rate_limit.message())
           .route(web::post().to(route_post::<AddAdmin>)),
+      )
+      // Reports
+      .service(
+        web::scope("/report")
+            .wrap(rate_limit.message())
+            .route("", web::get().to(route_get::<GetReportCount>))
+            .route("",web::post().to(route_post::<CreateReport>))
+            .route("/resolve",web::put().to(route_post::<ResolveReport>))
+            .route("/list", web::get().to(route_get::<ListReports>))
```

Comment:
As stated above the reporting / resolving should be moved into item field IE `/comment/report`

What do you think these should be at @Nutomic

Reply:
These look fine to me.

Reply:
`/comment/report/list` etc sounds fine to me if we're going with separate endpoints. Or possibly `/report/comment/list` would be an option.

Reply:
Ya, the front end could just call both those, (it needs a switch for `Both, Comment, and Post` anyway), and it can combine them how it sees fit.

I like `/comment/report`, `/comment/report/list` best.